sui-execution/latest/sui-adapter/src/execution_engine.rs

```diff
@@ -201,6 +201,67 @@ mod checked {
         (inner, effects, execution_result)
     }
 
+    #[instrument(
+        name = "commit_transaction_without_execution",
+        level = "debug",
+        skip_all
+    )]
+    pub fn commit_transaction_without_execution<Mode: ExecutionMode>(
+        shared_object_refs: Vec<ObjectRef>,
+        mut temporary_store: TemporaryStore<'_>,
+        transaction_kind: TransactionKind,
+        transaction_signer: SuiAddress,
+        gas_charger: &mut GasCharger,
+        transaction_digest: TransactionDigest,
+        mut transaction_dependencies: BTreeSet<TransactionDigest>,
+        epoch_id: &EpochId,
+        enable_expensive_checks: bool,
+    ) -> (
+        InnerTemporaryStore,
+        TransactionEffects,
+        Result<Mode::ExecutionResults, ExecutionError>,
+    ) {
+        let error = ExecutionError::new(ExecutionErrorKind::CertificateDenied, None);
```

Comment:
`CertificateDenied` is explicitly reserved for denying a cert through a list of digests.
For using deleted shared object please create a new error for it.

Reply:
There was an effort on the devX side to consolidate error types, and instead let the error message carry the specific information to the user on what to do about the error

Reply:
CertificateDenied is very specific to a case where the blockchain denies a specific certificate due to security issues (e.g. DoS or crash). Feel free to use another existing error if you don't want to introduce a new error, but please don't use this one as it would be very confusing to us.